plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/TestPinotConnectorTest.java

```diff
@@ -0,0 +1,379 @@
+import static io.trino.plugin.pinot.PinotQueryRunner.createPinotQueryRunner;
+import static io.trino.plugin.pinot.TestingPinotCluster.PINOT_LATEST_IMAGE_NAME;
+import static io.trino.spi.type.VarcharType.VARCHAR;
+import static io.trino.testing.MaterializedResult.resultBuilder;
+import static java.time.temporal.ChronoUnit.SECONDS;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestPinotConnectorTest
+        extends BaseConnectorTest
+{
+    // Use a recent value for updated_at to ensure Pinot doesn't clean up records older than retentionTimeValue as defined in the table specs
+    private static final Instant initialUpdatedAt = Instant.now().minus(Duration.ofDays(1)).truncatedTo(SECONDS);
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        TestingKafka kafka = closeAfterClass(TestingKafka.createWithSchemaRegistry());
+        kafka.start();
+        TestingPinotCluster pinot = closeAfterClass(new TestingPinotCluster(kafka.getNetwork(), isSecured(), getPinotImageName()));
+        pinot.start();
+
+        DistributedQueryRunner queryRunner = createPinotQueryRunner(
+                ImmutableMap.of(),
+                pinotProperties(pinot),
+                Optional.of(binder -> newOptionalBinder(binder, PinotHostMapper.class).setBinding()
+                        .toInstance(new TestingPinotHostMapper(pinot.getBrokerHostAndPort(), pinot.getServerHostAndPort(), pinot.getServerGrpcHostAndPort()))));
+
+        queryRunner.installPlugin(new TpchPlugin());
+        queryRunner.createCatalog("tpch", "tpch");
+
+        // We need the query runner to populate nation and region data from tpch schema
+        createAndPopulateNationAndRegionData(kafka, pinot, queryRunner);
+
+        return queryRunner;
+    }
+
+    private Map<String, String> pinotProperties(TestingPinotCluster pinot)
+    {
+        return ImmutableMap.<String, String>builder()
+                .put("pinot.controller-urls", pinot.getControllerConnectString())
+                //.put("pinot.max-rows-per-split-for-segment-queries", String.valueOf(MAX_ROWS_PER_SPLIT_FOR_SEGMENT_QUERIES))
+                //.put("pinot.max-rows-for-broker-queries", String.valueOf(MAX_ROWS_PER_SPLIT_FOR_BROKER_QUERIES))
+                .putAll(additionalPinotProperties())
+                .buildOrThrow();
+    }
+
+    protected Map<String, String> additionalPinotProperties()
+    {
+        if (isGrpcEnabled()) {
+            return ImmutableMap.of("pinot.grpc.enabled", "true");
+        }
+        return ImmutableMap.of();
+    }
+
+    protected boolean isGrpcEnabled()
+    {
+        return true;
+    }
+
+    protected boolean isSecured()
+    {
+        return false;
+    }
+
+    protected String getPinotImageName()
+    {
+        return PINOT_LATEST_IMAGE_NAME;
+    }
+
+    private void createAndPopulateNationAndRegionData(TestingKafka kafka, TestingPinotCluster pinot, DistributedQueryRunner queryRunner)
```

Comment:
nit: Rename to `copyTpchTables` for consistency with other connectors.

Reply:
Can we move this method to PinotQueryRunner so that we can use it from `main` method?

Reply:
Good idea! done.